### DISCUSSION TITLE
best delay settings

### DISCUSSION THREAD

**Discussion by Gregmanz** [victornpb/undiscord#414] [1. Questions & Answers]
What is the best delay setting that's the fastest deletion you can do without getting rate limited?

Im using 4000 delete delay and 250 search delay and it works but wondering if there's a faster one

**Comment by abigailwillow** [comment]


**Reply by Gregmanz** [reply]
> Search delay doesn't matter too much. I'd suggest 2500 delete delay and 100 search delay as in #402.

okay, but I've tried 3000 and it would still rate limit. I'm not too sure how much rate limit matters but would this be better?

**Reply by Gregmanz** [reply]
ima try that when i get home thanks

**Reply by Gregmanz** [reply]
> > okay, but I've tried 3000 and it would still rate limit. I'm not too sure how much rate limit matters but would this be better?
> 
> If you're deleting your own messages 2500 ms should be more than enough. Rate limiting is a lot more strict for deleting others' messages.

ok so what would be best for deleting others messages

**Reply by Gregmanz** [reply]
> > ok so what would be best for deleting others messages
> 
> I'm sorry, I'm not too sure about that since I mainly use the script for deleting my own messages. However, you could try purposefully setting it too low and waiting for Discord to rate limit you and try the delay that Discord's API initially times you out for. Hope that helps!

ok thanks

**Reply by Gregmanz** [reply]
ok figured out that best for deleting others is 3850 (delete delay) and 100 (search delay) its been going for 300+ messages and it hasn't been rate limited

**Reply by Gregmanz** [reply]
3800 and below isnt good

**Comment by ofernandofilo** [comment]
I performed a series of tests... here is my results here!
```
[search] + 25*[delete] = full time delay for 25 deleted msgs blocks
   12800 + 25* 1600    = 52800 [api warning about 200 msgs deleted]
    6400 + 25* 1900    = 53900 [api warning about 200 msgs deleted]
    4200 + 25* 2000    = 54200 [api warning about 200 msgs deleted] 
 >  2100 + 25* 2100    = 54600 [api warning about 150 msgs deleted]
 >  2100 + 25* 2100    = 54600 [tested 10K+, my new best!]
    2400 + 25* 2100    = 54900 [tested 999+, OK!]
     100 + 25* 2200    = 55100 [tested 700+, OK!]
     200 + 25* 2200    = 55200 [tested 800+, OK!]
     500 + 25* 2200    = 55500 [tested 999+, OK!]
    1000 + 25* 2200    = 56000 [tested 999+, OK!]
    6300 + 25* 2000    = 56300 [api warning about 150 msgs deleted]
    6400 + 25* 2000    = 56400 [tested 10K+, OK!] (my old value)
    1400 + 25* 2200    = 56400 [tested 700+, OK!]
     100 + 25* 2300    = 57600 [not tested - but a little slower]
     100 + 25* 2500    = 62600 [not tested - but slower]  
     100 + 25* 3850    = 96350 [not tested - but much slower]

[my new best for 10K+ msgs]
     search delay: 2100
     delete delay: 2100
     # the accumulative time for blocks of 25 deleted messages is ~55s;
     # for under 55s I don't intend to make adjustments or tests anymore;
     # the best values produced different behaviors in different situations. 
          it both produced (a) API alert message, when in another situation, 
          with computer / discord completely abandoned for long period, 
          (b) deleted more than 10K msgs without alert;
     # tests performed with at least 1 minute break between them;
     # tests performed without any other simultaneous use of discord;
     # I suspect that typing or using discord but mainly performing new deletions while using
          undiscord with the present values will compromise the results presented;
     # I fear that the lowest times for erasure are dynamically controlled by the API... 
          if this is the case, at times of greater use of the discord service, the lowest 
          possible values may vary;
     # tested using: KDE neon (linux), Waterfox / Brave (browser), tampermonkey, undiscord 5.0.3;
     # 2022-10-30, Brazil [the best results was obtained on a Sunday morning]
```

[if the script doesn't run, try]
- disable EFF HTTPS Everywhere extension;
- open developer tools first [ctrl+shift+i or f12];
- change browser;

for a smaller amount of messages deleted in batch, surely the minimum values can be smaller. _o/

**about edits**: I performed more tests after publishing, and I'm done!

**Reply by ofernandofilo** [reply]
I tested displayed values... for over 500+ messages and the fastest option for batches of 25 deleted messages I found was this one.

somewhere I saw the suggestion to use 2500 ms to delete messages. however, this even without taking into account the "search delay" produces more than 60s to delete blocks of 25 messages. so it's less efficient than the value I found (mine value is smaller around just 6s, but smaller).

so, I use only 2000ms to delete, but with an additional search delay of 6400ms minimum.

cheers! _o/
